plugin-viewport/src/main/kotlin/com/mapbox/maps/plugin/viewport/ViewportPluginImpl.kt

```diff
@@ -105,9 +108,11 @@ class ViewportPluginImpl : ViewportPlugin {
     @Synchronized get
 
   private fun updateStatus(targetStatus: ViewportStatus, reason: String) {
-    val previousStatus = status
-    status = targetStatus
-    notifyStatusChanged(previousStatus, targetStatus, reason)
+    if (targetStatus != status) {
+      val previousStatus = status
```

Comment:
how many times this update can happen? 
if too frequent maybe move `previousStatus` outside and don't recreate it?

Reply:
it shouldn't be very frequent, will only happen if user triggers a new state transition.
Could you suggest more details how would you optimise it?

Reply:
just add `previousStatus` as variable for this class. GC wouldn't clear it as current local value